packages/grafana-ui/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { PureComponent, useContext, FC } from 'react';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { ThemeContext } from '../../themes/index';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '../../types/theme';
+import tinycolor from 'tinycolor2';
+import { ToggleButtonGroupProps, ToggleButtonProps, ToggleButtonState } from './types';
+import { increaseContrast } from '../../utils/colors';
+import { ButtonSize } from '../Button/AbstractButton';
+
+const ToggleButtonGroup: FC<ToggleButtonGroupProps> = ({ label, children, propsPriority }) => {
+  const styles = getStyles(useContext(ThemeContext));
+  return (
+    <div>
+      <span className={styles.label}>{label}</span>
+      {React.Children.map(children, (child: any, index) => {
+        // Set correct style depending on first/middle/last
+        const style = index === 0 ? styles.first : index === children.length - 1 ? styles.last : styles.middle;
+
+        return React.cloneElement(child, {
+          className: style,
+          propsPriority,
+        });
+      })}
+    </div>
+  );
+};
+
+ToggleButtonGroup.displayName = 'ToggleButtonGroup';
+
+export { ToggleButtonGroup };
+export class ToggleButton extends PureComponent<ToggleButtonProps, ToggleButtonState> {
+  constructor(props: ToggleButtonProps) {
+    super(props);
+    this.state = {
+      untouched: true,
+      selected: props.selected,
+      value: props.value,
+    };
+
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e: React.SyntheticEvent) {
+    e.stopPropagation();
+
+    this.setState(prevState => ({
+      untouched: false,
+      selected: this.props.propsPriority ? this.props.selected : !prevState.selected,
+    }));
+    if (this.props.onChange) {
+      this.props.onChange(this.state);
+    }
+  }
+
+  handleBlur() {
+    this.setState({
+      untouched: true,
+    });
+  }
+
+  componentDidUpdate(prevProps: ToggleButtonProps) {
+    //Override internal select toggle if props are updated
+    if (prevProps.selected !== this.props.selected) {
+      this.setState({
+        selected: this.props.selected,
+      });
+    }
+  }
+
+  renderButton(theme: GrafanaTheme) {
+    const styles = getStyles(theme, this.props.size);
+    const { children, className } = this.props;
+    const activeStyle = this.state.selected ? styles.active : null;
+
+    return (
+      <button
+        onBlur={this.handleBlur}
+        className={cx([styles.button, activeStyle, className, this.state.untouched ? styles.focus : styles.unFocused])}
+        onClick={this.handleClick}
+      >
+        <span>{children}</span>
+      </button>
+    );
+  }
+
+  render() {
+    const { tooltip } = this.props;
+
+    const button = <ThemeContext.Consumer>{theme => this.renderButton(theme)}</ThemeContext.Consumer>;
+
+    if (tooltip) {
+      return (
+        <Tooltip content={tooltip} placement="bottom">
```

Comment:
Ideally the tooltip should be implemented in an AbstractButton component so that we don't need to think about handling tooltip in multiple button implementations.